src/discussions/posts/post/PostBanner.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useContext } from 'react';
+
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { AppContext } from '@edx/frontend-platform/react';
+import { Alert } from '@edx/paragon';
+import {
+  Pin,
+  Flag,
+  VerifiedBadge,
+  MoreVert,
+} from '@edx/paragon/icons';
+
+import messages from './messages';
+import { postShape } from './Post';
+import { commentShape } from '../../comments/comment/Comment';
+
+function banner(post) {
+  if (post.abuseFlagged) {
+    return { variant: 'danger', icon: Flag, message: messages.contentReported };
+  }
+  if (post.hasClosed) {
+    return { variant: 'info', icon: MoreVert, message: messages.closed };
+  }
+  if (post.endorsed) {
+    return { variant: 'success', icon: VerifiedBadge, message: messages.endorsed };
+  }
+  if (post.pinned) {
+    return { variant: 'info', icon: Pin, message: messages.pinned };
+  }
+  return null;
+}
+
+function PostBanner({ post, intl }) {
+  const { authenticatedUser } = useContext(AppContext);
+  const bannerProps = banner(post);
+  // show pinned, reported, endorsed and closed posts for moderators
+  if (authenticatedUser.administrator && bannerProps) {
```

Comment:
You don't need to check this, if the props exist, render the component. They should only exist for the correct type of user.